src/atomic.rs

```diff
@@ -560,6 +575,39 @@ impl<T> Atomic<T> {
     }
 }
 
+impl<T> fmt::Debug for Atomic<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let data = self.data.load(Ordering::SeqCst);
+        let raw = (data & !low_bits::<T>()) as *const T;
+        let tag = data & low_bits::<T>();
+
+        f.debug_struct("Atomic")
+            .field("raw", &raw)
+            .field("tag", &tag)
+            .finish()
+    }
+}
+
+impl<T> fmt::Pointer for Atomic<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let data = self.data.load(Ordering::SeqCst);
+        let raw = (data & !low_bits::<T>()) as *const T;
+        fmt::Pointer::fmt(&raw, f)
+    }
+}
+
+impl<T> Clone for Atomic<T> {
+    fn clone(&self) -> Self {
+        // NOTE: If you're cloning `Atomic`s, then you're *probably* using them as shared immutable
+        // pointers on the heap (i.e. they won't be concurrently modified at the same time they're
+        // being cloned). For that reason we'd most likely be okay with a `Relaxed` load here.
+        // However, just to avoid potential subtle pitfalls, `SeqCst` is used instead as the most
+        // conservative and safest choice of ordering.
```

Comment:
I prefer `Relaxed` here, because `clone()` is not necessarily associated with any synchronization.  It's semantics would be copy **a** value of the atomic, but without synchronization it can be a stale one.  If the user wants some synchronization, s/he can do it with other atomic operations or fences.

Reply:
I agree. But then again, `Debug` implementation for `AtomicPtr` uses the `SeqCst` ordering, as you can see [here](https://doc.rust-lang.org/nightly/src/core/sync/atomic.rs.html#1811-1815). It could/should use `Relaxed` by the same reasoning, right? My guess is that `SeqCst` is used "just in case" because it's the safest option.

Reply:
I'm mixed for using `Relaxed` in `Debug`..

 On one hand, I'd like to say we should put `Relaxed` there and requires users to put `fence(SeqCst)` if they are not sure of orderings, since `Debug::fmt()` has nothing to do with synchronization.  The same logic for `clone()`.

On the other hand, `Debug::fmt()` is primarily for debugging purposes, as opposed to `fn clone()` which can be used in production, and for that using `SeqCst` is perfectly okay.  However, the use of `SeqCst` should not imply that `Debug::fmt()` is always using a "safe" ordering..

Reply:
That's fair enough. :) I'll switch to `Relaxed` ordering.